runtime/src/lib.rs

```diff
@@ -101,6 +103,61 @@ impl Runtime {
     }
 }
 
+struct EnvAdapter<'a>{
```

Comment:
The name sounds a bit funny here and may need some bikeshedding. How does `SystemEnvironment` sound?

I agree that we are using the adapter pattern (if you want to use the technical term for this), but you could also say it's an `Environment` backed by the runtime `System`.

Reply:
How about ```IAdapterFactoryFacade``` ? 

Just kidding. I'm good with ```SystemEnvironment```